addons/coupon/models/coupon_program.py

```diff
@@ -153,3 +146,14 @@ def _get_valid_products(self, products):
             domain = ast.literal_eval(self.rule_products_domain)
             return products.filtered_domain(domain)
         return products
+
+    def _get_discount_product_values(self):
+        return {
+            'name': self.reward_id.display_name,
+            'type': 'service',
+            'taxes_id': False,
+            'supplier_taxes_id': False,
+            'sale_ok': False,
+            'purchase_ok': False,
+            'lst_price': 0, #Do not set a high value to avoid issue with coupon code
+        }
```

Comment:
```suggestion
            'purchase_ok': False,
            'lst_price': 0, #Do not set a high value to avoid issue with coupon code
            'invoice_policy': 'order' # Discount products should always be invoiced on ordered quantities
        }
```